libbeat/outputs/tidb_slow_query/schema.go

```diff
@@ -0,0 +1,191 @@
+		"Wait_time":                     "double",
+		"Backoff_time":                  "double",
+		"LockKeys_time":                 "double",
+		"Request_count":                 "double",
+		"Total_keys":                    "double",
+		"Process_keys":                  "double",
+		"DB":                            "varchar(64)",
+		"Index_names":                   "varchar(128)",
+		"Digest":                        "varchar(64)",
+		"Stats":                         "varchar(512)",
+		"Cop_proc_avg":                  "double",
+		"Cop_proc_p90":                  "double",
+		"Cop_proc_max":                  "double",
+		"Cop_proc_addr":                 "varchar(64)",
+		"Cop_wait_avg":                  "double",
+		"Cop_wait_p90":                  "double",
+		"Cop_wait_max":                  "double",
+		"Cop_wait_addr":                 "varchar(64)",
+		"Mem_max":                       "double",
+		"Disk_max":                      "double",
+		"Prev_stmt":                     "longtext",
+		"Query":                         "longtext",
+		"Is_internal":                   "tinyint(1)",
+		"Succ":                          "tinyint(1)",
+		"Plan_from_cache":               "tinyint(1)",
+		"Plan_from_binding":             "tinyint(1)",
+		"Prepared":                      "tinyint(1)",
+		"KV_total":                      "double",
+		"PD_total":                      "double",
+		"Backoff_total":                 "double",
+		"Write_sql_response_total":      "double",
+		"Plan":                          "longtext",
+		"Plan_digest":                   "varchar(128)",
+		"Backoff_Detail":                "varchar(4096)",
+		"Rocksdb_delete_skipped_count":  "double",
+		"Rocksdb_key_skipped_count":     "double",
+		"Rocksdb_block_cache_hit_count": "double",
+		"Rocksdb_block_read_count":      "double",
+		"Rocksdb_block_read_byte":       "double",
+	}
+	orderedColumn = make([]string, 0, len(slowQuerySQLType))
+	zone, _       = time.LoadLocation("Asia/Shanghai")
+)
+
+func init() {
+	for k := range slowQuerySQLType {
+		orderedColumn = append(orderedColumn, k)
+	}
+}
+
+const dateFormat = "2006-01-02"
+
+func insertStmt(schema, table string) string {
+	cols := make([]string, 0, len(orderedColumn))
+	args := make([]string, 0, len(orderedColumn))
+	for _, c := range orderedColumn {
+		cols = append(cols, quote(c))
+		args = append(args, "?")
+	}
+	buf := new(bytes.Buffer)
+	buf.WriteString(fmt.Sprintf("INSERT INTO %s.%s (", quote(schema), quote(table)))
+	buf.WriteString(strings.Join(cols, ","))
+	buf.WriteString(") ")
+	buf.WriteString("VALUES (")
+	buf.WriteString(strings.Join(args, ","))
+	buf.WriteString(") ")
+	buf.WriteString(";")
+	return buf.String()
+}
+
+func createTableStmt(schema, table string, lessThanPartitions []time.Time) string {
+	buf := new(bytes.Buffer)
+	buf.WriteString(fmt.Sprintf("CREATE TABLE IF NOT EXISTS %s.%s (", quote(schema), quote(table)))
+	buf.WriteString("`id` bigint(20) unsigned not null AUTO_INCREMENT,")
```

Comment:
You can use `AUTO_RANDOM` instead of `AUTO_INCREMENT` in TiDB to avoid causing hotspots.

Reply:
I cannot. AUTO_RANDOM has limitation from cluster index and partition key. Please see https://docs.pingcap.com/tidb/stable/auto-random#restrictions